framework/db: use HikariCP as default and improvements

shwstppr opened the pull request:
### Description

Allows configuring connection pool library for database connection. As default, replaces dbcp2 connection pool library with more performant HikariCP.

`db.<DATABASE>.connectionPoolLib` property can be set in the `db.properties` to use the desired library.
Set `dbcp` for using DBCP2
Set `hikaricp` or <EMPTY> for using HikariCP

Per docs, if the mysql connector is JDBC2 compliant then it should use the Connection.isValid API to test a connection.
(https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#isValid-int-)

This would significantly reduce query lags and API throughput, as for every SQL query one or two SELECT 1 are performed every time a Connection is given to application logic.

This should only be accepted when the driver is JDBC4 complaint.

As per the docs, the connector-j can use /* ping */ before calling SELECT 1 to have light weight application pings to the server:
https://dev.mysql.com/doc/connector-j/en/connector-j-usagenotes-j2ee-concepts-connection-pooling.html

Related doc PR: https://github.com/apache/cloudstack-documentation/pull/431

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial


### Screenshots (if appropriate):


### How Has This Been Tested?

<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->

#### How did you try to break this feature and the system with this change?

<!-- see how your change affects other areas of the code, etc. -->


<!-- Please read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/main/CONTRIBUTING.md) document -->

shwstppr commented:
@blueorangutan package

blueorangutan commented:
@shwstppr a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10632

rohityadavcloud commented:
@blueorangutan package

blueorangutan commented:
@rohityadavcloud a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10730

shwstppr commented:
@blueorangutan test

blueorangutan commented:
<b>[SF] Trillian test result (tid-11125)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 50273 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11125-kvm-ol8.zip
Smoke tests completed. 128 look OK, 0 have errors, 11 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
all_test_vm_strict_host_tags | `Skipped` | --- | test_vm_strict_host_tags.py
all_test_vnf_templates | `Skipped` | --- | test_vnf_templates.py
all_test_volumes | `Skipped` | --- | test_volumes.py
all_test_vpc_ipv6 | `Skipped` | --- | test_vpc_ipv6.py
all_test_vpc_redundant | `Skipped` | --- | test_vpc_redundant.py
all_test_vpc_router_nics | `Skipped` | --- | test_vpc_router_nics.py
all_test_vpc_vpn | `Skipped` | --- | test_vpc_vpn.py
all_test_webhook_delivery | `Skipped` | --- | test_webhook_delivery.py
all_test_webhook_lifecycle | `Skipped` | --- | test_webhook_lifecycle.py
all_test_host_maintenance | `Skipped` | --- | test_host_maintenance.py
all_test_hostha_kvm | `Skipped` | --- | test_hostha_kvm.py

rohityadavcloud commented on framework/db/src/main/java/com/cloud/utils/db/TransactionLegacy.java:
```diff
@@ -1226,45 +1220,84 @@ private static DataSource createDataSource(String uri, String username, String p
+        config.setConnectionTimeout(30000); // 30 seconds in milliseconds
+        config.setKeepaliveTime(600000);    // Keepalive time in milliseconds (10 minutes)
+        config.setIdleTimeout(300000); // 5 minutes
+        //config.setMinimumIdle(maxIdle);
```
@shwstppr worth checking & removing dead/commented code; remove old variables in db.properties and or refactor/remove them to introduce for use with these configs ^^ which are hard-coded at the moment.

shwstppr commented:
@blueorangutan package

blueorangutan commented:
@shwstppr a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10826

blueorangutan commented:
@shwstppr a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

blueorangutan commented:
<b>[SF] Trillian test result (tid-11202)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 63749 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11202-kvm-ol8.zip
Smoke tests completed. 138 look OK, 1 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Error` | 771.12 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | `Error` | 1144.52 | test_vpc_redundant.py

shwstppr commented:
@blueorangutan package

blueorangutan commented:
@shwstppr a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10861

shwstppr commented:
@rohityadavcloud I've addressed your comment and added changes to configure the connection pool library using `db.properties`. (details in the description)

@blueorangutan package

blueorangutan commented:
@shwstppr a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10863

shwstppr commented:
@blueorangutan test

blueorangutan commented:
@shwstppr a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

DaanHoogland commented on framework/db/src/main/java/com/cloud/utils/db/TransactionLegacy.java:
```diff
@@ -1020,6 +1025,21 @@ public static void initDataSource(String propsFileName) throws IOException {
+            return null;
+        }
+        try {
+            if (type.equals(Long.class)) {
+                return type.cast(Long.parseLong(value));
+            } else {
+                return type.cast(Integer.parseInt(value));
+            }
```
we are never expected to use float/double or some other format?

DaanHoogland reviewed (approved):
clgtm,
question about long term maintenance; this will only work for known types of connection pools due to the `createDataSource()` implementation (and nothing else), right?

blueorangutan commented:
<b>[SF] Trillian test result (tid-11235)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 61543 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11235-kvm-ol8.zip
Smoke tests completed. 131 look OK, 2 have errors, 6 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_08_upgrade_kubernetes_ha_cluster | `Failure` | 475.80 | test_kubernetes_clusters.py
test_03_secured_to_nonsecured_vm_migration | `Error` | 348.20 | test_vm_life_cycle.py
all_test_volumes | `Skipped` | --- | test_volumes.py
all_test_vpc_ipv6 | `Skipped` | --- | test_vpc_ipv6.py
all_test_vpc_redundant | `Skipped` | --- | test_vpc_redundant.py
all_test_vpc_router_nics | `Skipped` | --- | test_vpc_router_nics.py
all_test_vpc_vpn | `Skipped` | --- | test_vpc_vpn.py
all_test_webhook_delivery | `Skipped` | --- | test_webhook_delivery.py

shwstppr commented:
@blueorangutan test matrix

blueorangutan commented:
<b>[SF] Trillian Build Failed (tid-11253)<b/>

shwstppr commented:
Created a documentation PR: https://github.com/apache/cloudstack-documentation/pull/431

blueorangutan commented:
<b>[SF] Trillian test result (tid-11251)</b>
Environment: kvm-ubuntu22 (x2), Advanced Networking with Mgmt server u22
Total time taken: 53220 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11251-kvm-ubuntu22.zip
Smoke tests completed. 138 look OK, 1 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_03_ping_in_ssvm_success | `Failure` | 15.53 | test_diagnostics.py
test_05_ping_in_cpvm_success | `Failure` | 15.53 | test_diagnostics.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11252)</b>
Environment: vmware-70u3 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 59468 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11252-vmware-70u3.zip
Smoke tests completed. 137 look OK, 2 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_deployVMInSharedNetwork | `Error` | 132.29 | test_network.py
test_02_restore_vm_with_disk_offering | `Error` | 58.09 | test_restore_vm.py
test_03_restore_vm_with_disk_offering_custom_size | `Error` | 61.17 | test_restore_vm.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11250)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 59833 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11250-kvm-ol8.zip
Smoke tests completed. 136 look OK, 3 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
ContextSuite context=TestClusterDRS>:setup | `Error` | 0.00 | test_cluster_drs.py
test_01_secure_vm_migration | `Error` | 378.52 | test_vm_life_cycle.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Failure` | 406.40 | test_vpc_redundant.py

shwstppr commented:
@blueorangutan test

blueorangutan commented:
@shwstppr a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

blueorangutan commented:
<b>[SF] Trillian test result (tid-11276)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 62482 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11276-kvm-ol8.zip
Smoke tests completed. 134 look OK, 1 have errors, 4 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
ContextSuite context=TestRVPCSite2SiteVpn>:setup | `Error` | 0.00 | test_vpc_vpn.py
ContextSuite context=TestVPCSite2SiteVPNMultipleOptions>:setup | `Error` | 0.00 | test_vpc_vpn.py
ContextSuite context=TestVpcRemoteAccessVpn>:setup | `Error` | 0.00 | test_vpc_vpn.py
ContextSuite context=TestVpcSite2SiteVpn>:setup | `Error` | 0.00 | test_vpc_vpn.py
all_test_webhook_delivery | `Skipped` | --- | test_webhook_delivery.py
all_test_webhook_lifecycle | `Skipped` | --- | test_webhook_lifecycle.py
all_test_host_maintenance | `Skipped` | --- | test_host_maintenance.py
all_test_hostha_kvm | `Skipped` | --- | test_hostha_kvm.py

shwstppr commented:
@blueorangutan package

blueorangutan commented:
@shwstppr a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

rohityadavcloud commented on framework/db/src/main/java/com/cloud/utils/db/TransactionLegacy.java:
```diff
@@ -95,6 +98,8 @@ public class TransactionLegacy implements Closeable {
         }
     }
 
+    private static final String CONNECTION_POOL_LIB_DBCP = "dbcp";
```
@shwstppr minor - can we change the default to hikaricp, so when not defined hikaricp is used; otherwise we honour one of which is defined in db.properties ?

shwstppr replied:
@rohityadavcloud we are already doing that in `createDataSource` and `getDefaultDataSource`

rohityadavcloud reviewed (approved):
LGTM, left a minor comment.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10929

shwstppr commented:
@blueorangutan test

blueorangutan commented:
@shwstppr a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

rohityadavcloud commented:
@blueorangutan test ol8 vmware-70u3

weizhouapache commented:
@shwstppr 
anything to do for upgrade ?

does `hikaricp` support most popular databases ?

shwstppr commented:
> @shwstppr anything to do for upgrade ?
> 
No, I was able touse either library just by changing db.properties. But will still do some upgrade tests to rule out any regressions

> does `hikaricp` support most popular databases ?

Yes, https://github.com/brettwooldridge/HikariCP?tab=readme-ov-file#popular-datasource-class-names

weizhouapache commented on framework/db/src/main/java/com/cloud/utils/db/TransactionLegacy.java:
```diff
@@ -1267,6 +1362,44 @@ private static GenericObjectPoolConfig createPoolConfig(Integer maxActive, Integ
+    private static DataSource getDefaultDataSource(final String connectionPoolLib, final String database) {
+        LOGGER.debug("Creating default datasource for database: {} with connection pool lib: {}",
+                database, connectionPoolLib);
+        if (CONNECTION_POOL_LIB_DBCP.equals(connectionPoolLib)) {
```
equalsIgnoreCase ?

rohityadavcloud replied:
```suggestion
        if (CONNECTION_POOL_LIB_DBCP. equalsIgnoreCase(connectionPoolLib)) {
```

weizhouapache reviewed (approved):
thanks @shwstppr 
code lgtm

blueorangutan commented:
<b>[SF] Trillian test result (tid-11316)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 64859 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11316-kvm-ol8.zip
Smoke tests completed. 133 look OK, 2 have errors, 4 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_nonsecured_to_secured_vm_migration | `Error` | 376.97 | test_vm_life_cycle.py
test_01_vpc_remote_access_vpn | `Error` | 309.28 | test_vpc_vpn.py
ContextSuite context=TestVpcRemoteAccessVpn>:teardown | `Error` | 1065.21 | test_vpc_vpn.py
test_01_vpc_site2site_vpn | `Failure` | 1621.20 | test_vpc_vpn.py
test_01_vpc_site2site_vpn | `Error` | 1621.21 | test_vpc_vpn.py
ContextSuite context=TestVpcSite2SiteVpn>:teardown | `Error` | 1759.09 | test_vpc_vpn.py
all_test_webhook_delivery | `Skipped` | --- | test_webhook_delivery.py
all_test_webhook_lifecycle | `Skipped` | --- | test_webhook_lifecycle.py
all_test_host_maintenance | `Skipped` | --- | test_host_maintenance.py
all_test_hostha_kvm | `Skipped` | --- | test_hostha_kvm.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11338)</b>
Environment: vmware-70u3 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 44280 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9518-t11338-vmware-70u3.zip
Smoke tests completed. 101 look OK, 29 have errors, 9 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_deployVMInSharedNetwork | `Error` | 138.11 | test_network.py
test_01_vpc_privategw_acl | `Failure` | 953.26 | test_privategw_acl.py
test_01_vpc_privategw_acl | `Error` | 953.27 | test_privategw_acl.py
test_02_vpc_privategw_static_routes | `Error` | 43.09 | test_privategw_acl.py
test_03_vpc_privategw_restart_vpc_cleanup | `Error` | 48.14 | test_privategw_acl.py
test_04_rvpc_privategw_static_routes | `Error` | 54.94 | test_privategw_acl.py
ContextSuite context=TestPrivateGwACL>:teardown | `Error` | 92.57 | test_privategw_acl.py
test_04_deploy_vnf_appliance | `Error` | 21.08 | test_vnf_templates.py
test_04_deploy_vnf_appliance | `Error` | 21.08 | test_vnf_templates.py
test_05_delete_vnf_template | `Error` | 0.03 | test_vnf_templates.py
ContextSuite context=TestVnfTemplates>:teardown | `Error` | 0.09 | test_vnf_templates.py
ContextSuite context=TestCreateVolume>:setup | `Error` | 0.00 | test_volumes.py
ContextSuite context=TestVolumes>:setup | `Error` | 2.85 | test_volumes.py
ContextSuite context=TestIpv6Vpc>:setup | `Error` | 0.00 | test_vpc_ipv6.py
test_02_restore_vm_with_disk_offering | `Error` | 58.14 | test_restore_vm.py
test_03_restore_vm_with_disk_offering_custom_size | `Error` | 55.05 | test_restore_vm.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Error` | 3.02 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | `Error` | 2.98 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | `Error` | 2.76 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | `Error` | 2.78 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 2.85 | test_vpc_redundant.py
test_01_VPC_nics_after_destroy | `Error` | 3.10 | test_vpc_router_nics.py
test_02_VPC_default_routes | `Error` | 3.16 | test_vpc_router_nics.py
test_01_redundant_vpc_site2site_vpn | `Error` | 0.06 | test_vpc_vpn.py
ContextSuite context=TestRVPCSite2SiteVpn>:teardown | `Error` | 0.08 | test_vpc_vpn.py
test_01_vpc_site2site_vpn_multiple_options | `Error` | 0.04 | test_vpc_vpn.py
ContextSuite context=TestVPCSite2SiteVPNMultipleOptions>:teardown | `Error` | 0.07 | test_vpc_vpn.py
test_01_vpc_remote_access_vpn | `Failure` | 0.07 | test_vpc_vpn.py
ContextSuite context=TestVpcRemoteAccessVpn>:teardown | `Error` | 0.09 | test_vpc_vpn.py
test_01_vpc_site2site_vpn | `Error` | 0.04 | test_vpc_vpn.py
ContextSuite context=TestVpcSite2SiteVpn>:teardown | `Error` | 0.06 | test_vpc_vpn.py
test_01_prepare_and_cancel_shutdown | `Error` | 2.46 | test_safe_shutdown.py
test_02_trigger_shutdown | `Failure` | 301.40 | test_safe_shutdown.py
test_01_scale_vm | `Error` | 0.02 | test_scale_vm.py
test_02_scale_vm_negative_offering_disable_scaling | `Error` | 0.02 | test_scale_vm.py
test_03_scale_vm_negative_vm_disable_scaling | `Error` | 0.02 | test_scale_vm.py
test_04_scale_vm_with_user_account | `Error` | 2.81 | test_scale_vm.py
test_04_scale_vm_with_user_account | `Error` | 2.82 | test_scale_vm.py
test_05_scale_vm_dont_allow_disk_offering_change | `Error` | 0.09 | test_scale_vm.py
ContextSuite context=TestScaleVm>:teardown | `Error` | 0.22 | test_scale_vm.py
test_01_webhook_deliveries | `Error` | 4.92 | test_webhook_delivery.py
ContextSuite context=TestServiceOfferings>:setup | `Error` | 3.22 | test_service_offerings.py
test_01_create_webhook_admin_local | `Error` | 0.12 | test_webhook_lifecycle.py
test_02_create_webhook_admin_domain | `Error` | 0.15 | test_webhook_lifecycle.py
test_03_create_webhook_admin_global | `Error` | 0.13 | test_webhook_lifecycle.py
test_04_create_webhook_domainadmin_local | `Error` | 3.41 | test_webhook_lifecycle.py
test_05_create_webhook_domainadmin_subdomain | `Error` | 3.30 | test_webhook_lifecycle.py
test_06_create_webhook_domainadmin_global_negative | `Error` | 3.17 | test_webhook_lifecycle.py
test_07_create_webhook_user_local | `Error` | 2.87 | test_webhook_lifecycle.py
test_08_create_webhook_user_domain | `Error` | 2.84 | test_webhook_lifecycle.py
test_09_create_webhook_user_gloabl | `Error` | 2.75 | test_webhook_lifecycle.py
test_10_create_webhook_admin_advanced | `Error` | 2.91 | test_webhook_lifecycle.py
test_11_update_webhook | `Error` | 2.93 | test_webhook_lifecycle.py
test_12_list_user_webhook_deliveries | `Error` | 7.91 | test_webhook_lifecycle.py
test_13_webhook_execute_delivery | `Error` | 3.23 | test_webhook_lifecycle.py
ContextSuite context=TestSetSourceNatIp>:setup | `Error` | 0.00 | test_set_sourcenat.py
test_01_cancel_host_maintenace_with_no_migration_jobs | `Error` | 7.65 | test_host_maintenance.py
test_02_cancel_host_maintenace_with_migration_jobs | `Error` | 0.35 | test_host_maintenance.py
test_03_cancel_host_maintenace_with_migration_jobs_failure | `Error` | 3.25 | test_host_maintenance.py
ContextSuite context=TestSnapshotRootDisk>:setup | `Error` | 0.00 | test_snapshots.py
ContextSuite context=TestSnapshotStandaloneBackup>:setup | `Error` | 0.00 | test_snapshots.py
test_05_stop_ssvm | `Error` | 0.13 | test_ssvm.py
test_06_stop_cpvm | `Error` | 0.11 | test_ssvm.py
test_07_reboot_ssvm | `Error` | 0.12 | test_ssvm.py
test_08_reboot_cpvm | `Error` | 0.11 | test_ssvm.py
test_09_reboot_ssvm_forced | `Error` | 0.13 | test_ssvm.py
test_10_reboot_cpvm_forced | `Error` | 0.13 | test_ssvm.py
test_11_destroy_ssvm | `Error` | 0.09 | test_ssvm.py
test_12_destroy_cpvm | `Error` | 0.08 | test_ssvm.py
test_01_import_storage_policies | `Error` | 4.38 | test_storage_policy.py
ContextSuite context=TestVMWareStoragePolicies>:teardown | `Error` | 4.42 | test_storage_policy.py
ContextSuite context=TestCopyAndDeleteTemplatesAcrossZones>:setup | `Error` | 0.00 | test_templates.py
ContextSuite context=TestCreateTemplate>:setup | `Error` | 0.00 | test_templates.py
ContextSuite context=TestTemplates>:setup | `Error` | 5.76 | test_templates.py
test_01_positive_tests_usage | `Failure` | 3.04 | test_usage_events.py
test_01_positive_tests_usage | `Error` | 3.04 | test_usage_events.py
test_01_ISO_usage | `Error` | 0.03 | test_usage.py
ContextSuite context=TestISOUsage>:teardown | `Error` | 0.05 | test_usage.py
ContextSuite context=TestLBRuleUsage>:setup | `Error` | 2.91 | test_usage.py
ContextSuite context=TestNatRuleUsage>:setup | `Error` | 5.83 | test_usage.py
ContextSuite context=TestPublicIPUsage>:setup | `Error` | 8.91 | test_usage.py
ContextSuite context=TestSnapshotUsage>:setup | `Error` | 12.61 | test_usage.py
ContextSuite context=TestTemplateUsage>:setup | `Error` | 15.81 | test_usage.py
ContextSuite context=TestVmUsage>:setup | `Error` | 18.58 | test_usage.py
ContextSuite context=TestVolumeUsage>:setup | `Error` | 21.50 | test_usage.py
ContextSuite context=TestVpnUsage>:setup | `Error` | 24.53 | test_usage.py
ContextSuite context=TestVmAutoScaling>:setup | `Error` | 0.00 | test_vm_autoscaling.py
test_01_deploy_vm_on_specific_host | `Error` | 0.08 | test_vm_deployment_planner.py
test_02_deploy_vm_on_specific_cluster | `Error` | 0.07 | test_vm_deployment_planner.py
test_03_deploy_vm_on_specific_pod | `Error` | 0.10 | test_vm_deployment_planner.py
test_04_deploy_vm_on_host_override_pod_and_cluster | `Error` | 0.12 | test_vm_deployment_planner.py
test_05_deploy_vm_on_cluster_override_pod | `Error` | 0.08 | test_vm_deployment_planner.py
ContextSuite context=TestVMDeploymentPlanner>:teardown | `Error` | 0.13 | test_vm_deployment_planner.py
ContextSuite context=TestDeployVM>:setup | `Error` | 0.00 | test_vm_life_cycle.py
ContextSuite context=TestKVMLiveMigration>:teardown | `Error` | 0.03 | test_vm_life_cycle.py
test_01_offline_migrate_VM_and_root_volume | `Error` | 0.04 | test_vm_life_cycle.py
test_02_offline_migrate_VM_with_two_data_disks | `Error` | 0.04 | test_vm_life_cycle.py
test_03_live_migrate_VM_with_two_data_disks | `Error` | 0.05 | test_vm_life_cycle.py
test_04_migrate_detached_volume | `Error` | 0.04 | test_vm_life_cycle.py
ContextSuite context=TestMigrateVMwithVolume>:teardown | `Error` | 0.07 | test_vm_life_cycle.py
test_01_vapps_vm_cycle | `Error` | 0.29 | test_vm_life_cycle.py
ContextSuite context=TestVAppsVM>:teardown | `Error` | 0.31 | test_vm_life_cycle.py
ContextSuite context=TestVMLifeCycle>:setup | `Error` | 3.13 | test_vm_life_cycle.py
test_01_unmanage_vm_cycle | `Error` | 0.59 | test_vm_lifecycle_unmanage_import.py
ContextSuite context=TestUnmanageVM>:teardown | `Error` | 0.62 | test_vm_lifecycle_unmanage_import.py
test_change_service_offering_for_vm_with_snapshots | `Error` | 0.02 | test_vm_snapshots.py
ContextSuite context=TestChangeServiceOfferingForVmWithSnapshots>:teardown | `Error` | 0.10 | test_vm_snapshots.py
ContextSuite context=TestVmSnapshot>:setup | `Error` | 2.82 | test_vm_snapshots.py
ContextSuite context=TestVMSchedule>:setup | `Error` | 0.00 | test_vm_schedule.py
test_01_migrate_vm_strict_tags_success | `Error` | 1.06 | test_vm_strict_host_tags.py
test_02_migrate_vm_strict_tags_failure | `Error` | 1.06 | test_vm_strict_host_tags.py
ContextSuite context=TestMigrateVMStrictTags>:teardown | `Error` | 2.76 | test_vm_strict_host_tags.py
test_01_restore_vm_strict_tags_success | `Error` | 1.07 | test_vm_strict_host_tags.py
test_02_restore_vm_strict_tags_failure | `Error` | 1.10 | test_vm_strict_host_tags.py
ContextSuite context=TestRestoreVMStrictTags>:teardown | `Error` | 2.04 | test_vm_strict_host_tags.py
test_01_scale_vm_strict_tags_success | `Error` | 0.97 | test_vm_strict_host_tags.py
test_02_scale_vm_strict_tags_failure | `Error` | 1.68 | test_vm_strict_host_tags.py
ContextSuite context=TestScaleVMStrictTags>:teardown | `Error` | 2.58 | test_vm_strict_host_tags.py
test_01_deploy_vm_on_specific_host_without_strict_tags | `Error` | 1.00 | test_vm_strict_host_tags.py
test_02_deploy_vm_on_any_host_without_strict_tags | `Error` | 0.99 | test_vm_strict_host_tags.py
test_03_deploy_vm_on_specific_host_with_strict_tags_success | `Error` | 0.97 | test_vm_strict_host_tags.py
test_04_deploy_vm_on_any_host_with_strict_tags_success | `Error` | 0.96 | test_vm_strict_host_tags.py
test_05_deploy_vm_on_specific_host_with_strict_tags_failure | `Failure` | 1.02 | test_vm_strict_host_tags.py
test_06_deploy_vm_on_any_host_with_strict_tags_failure | `Failure` | 1.05 | test_vm_strict_host_tags.py
ContextSuite context=TestVMDeploymentPlannerStrictTags>:teardown | `Error` | 2.02 | test_vm_strict_host_tags.py
all_test_private_roles | `Skipped` | --- | test_private_roles.py
all_test_projects | `Skipped` | --- | test_projects.py
all_test_public_ip_range | `Skipped` | --- | test_public_ip_range.py
all_test_purge_expunged_vms | `Skipped` | --- | test_purge_expunged_vms.py
all_test_pvlan | `Skipped` | --- | test_pvlan.py
all_test_quarantined_ips | `Skipped` | --- | test_quarantined_ips.py
all_test_regions | `Skipped` | --- | test_regions.py
all_test_register_userdata | `Skipped` | --- | test_register_userdata.py
all_test_reset_configuration_settings | `Skipped` | --- | test_reset_configuration_settings.py

rohityadavcloud commented:
LGTM, considering this and previous smoketests - intermittent failures observed, not likely from this PR. Merging based on this & the review lgtms.
